lib/kernel.coffee

```diff
@@ -11,20 +11,18 @@ WatchSidebar = require './watch-sidebar'
 
 module.exports =
 class Kernel
-    constructor: (@kernelSpec, @grammar, @config, @configPath, @onlyConnect = false) ->
+    constructor: (@onDestroy, @kernelName, @kernelSpec, filePath, @config, @configPath, @onlyConnect = false) ->
```

Comment:
we need to reduce the number of arguments this constructor is taking.

I think `filePath` is missing a `@`